src/pytorch_lightning/tuner/tuning.py

```diff
@@ -212,3 +263,34 @@ def lr_find(
         )
         self.trainer.auto_lr_find = False
         return result["lr_find"]
+
+
+def _check_tuner_configuration(
+    trainer: "pl.Trainer",
+    train_dataloaders: Optional[Union[TRAIN_DATALOADERS, "pl.LightningDataModule"]] = None,
+    val_dataloaders: Optional[EVAL_DATALOADERS] = None,
+    dataloaders: Optional[EVAL_DATALOADERS] = None,
+    method: str = "fit",
+) -> None:
+    supported_methods = ("fit", "validate", "test", "predict")
+    if method not in supported_methods:
+        raise MisconfigurationException(f"method {method!r} is invalid. Should be one of {supported_methods}.")
```

Comment:
Let's avoid using MisconfigurationException and use more standard exceptions (ValueError for the ones that check arguments) instead: #10918

Reply:
done!

Reply:
There are other instances that should be updated